pal/input/native/gamepad-input.ts

```diff
@@ -1,7 +1,374 @@
-export class GamepadInputSource {
-    support: boolean;
+/* eslint-disable brace-style */
+import { GamepadCallback } from 'pal/input';
+import { systemInfo } from 'pal/system-info';
+import { InputEventType } from '../../../cocos/input/types/event-enum';
+import { Feature } from '../../system-info/enum-type';
+import { EventTarget } from '../../../cocos/core/event/event-target';
+import { EventGamepad } from '../../../cocos/input/types';
+import { InputSourceButton, InputSourceDpad, InputSourceStick } from '../input-source';
+import { fastRemoveAt } from '../../../cocos/core/utils/array';
 
-    constructor () {
-        this.support = false;
+enum Button {
+    BUTTON_SOUTH,
+    BUTTON_EAST,
+    BUTTON_WEST,
+    BUTTON_NORTH,
+    NS_MINUS,
+    NS_PLUS,
+    BUTTON_L1,
+    BUTTON_L2,
+    BUTTON_L3,
+    BUTTON_R1,
+    BUTTON_R2,
+    BUTTON_R3,
+    DPAD_UP,
+    DPAD_DOWN,
+    DPAD_LEFT,
+    DPAD_RIGHT,
+    BUTTON_LEFT_STICK,
+    LEFT_STICK_UP,
+    LEFT_STICK_DOWN,
+    LEFT_STICK_LEFT,
+    LEFT_STICK_RIGHT,
+    BUTTON_RIGHT_STICK,
+    RIGHT_STICK_UP,
+    RIGHT_STICK_DOWN,
+    RIGHT_STICK_LEFT,
+    RIGHT_STICK_RIGHT,
```

Comment:
```suggestion
    LEFT_STICK_UP,
    LEFT_STICK_DOWN,
    LEFT_STICK_LEFT,
    LEFT_STICK_RIGHT,
    RIGHT_STICK_UP,
    RIGHT_STICK_DOWN,
    RIGHT_STICK_LEFT,
    RIGHT_STICK_RIGHT,
```

I think `BUTTON_LEFT_STICK` `BUTTON_RIGHT_STICK` is `BUTTON_L3` `BUTTON_R3`